pilot/pkg/model/jwks_resolver.go

```diff
@@ -384,41 +373,49 @@ func (r *JwksResolver) refresher() {
 	for {
 		select {
 		case <-r.refreshTicker.C:
-			currentHasError := r.refresh()
-			if currentHasError {
-				if lastHasError {
-					// update to exponential backoff if last time also failed.
-					r.refreshInterval *= 2
-					if r.refreshInterval > JwtPubKeyRefreshIntervalOnFailureResetThreshold {
-						r.refreshInterval = JwtPubKeyRefreshIntervalOnFailureResetThreshold
-					}
-				} else {
-					// change to the refreshIntervalOnFailure if failed for the first time.
-					r.refreshInterval = r.refreshIntervalOnFailure
-				}
-			} else {
-				// reset the refresh interval if success.
-				r.refreshInterval = r.refreshDefaultInterval
-			}
-			lastHasError = currentHasError
-			r.refreshTicker.Reset(r.refreshInterval)
+			lastHasError = r.refreshCache(lastHasError)
 		case <-closeChan:
 			r.refreshTicker.Stop()
 			return
+		case jwtKeyData := <-jwksuriChannel:
+			lastHasError = r.refreshCache(lastHasError)
+			if lastHasError {
```

Comment:
I think you do not need to log this anymore, it should already be logged inside the refresh().